lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -665,14 +678,140 @@ def write(chunk)
           rescue Google::Apis::ClientError => error
             # Most ClientErrors indicate a problem with the request itself and
             # should not be retried.
-            increment_dropped_entries_count(entries_count)
-            @log.warn "Dropping #{entries_count} log message(s)",
-                      error_class: error.class.to_s, error: error.to_s
+            error_details_map = {}
+            if @partial_success
+              error_details_map = construct_error_details_map(
+                extract_log_entry_errors(error))
+            end
+            if error_details_map.empty?
+              increment_dropped_entries_count(entries_count)
+              @log.warn "Dropping #{entries_count} log message(s)",
+                        error_class: error.class.to_s, error: error.to_s
+            else
+              error_details_map.each do |(error_code, error_message), indexes|
+                partial_error_count = indexes.length
+                increment_dropped_entries_count(partial_error_count)
+                @log.warn "Dropping #{partial_error_count} log message(s)",
+                          error_code: "google.rpc.Code.#{error_code}",
```

Comment:
Careful -- these are numeric codes, not symbolic names.

Reply:
Yeah, I'm aware of that. Do we want to do the translation here? Or change this to something like `"google.rpc.Code.[#{error_code}]"`

Reply:
Let's go with `"google.rpc.Code[#{error_code}]"`.

Reply:
Changed.